src/test/shell/bazel/bazel_coverage_hermetic_py_test.sh

```diff
@@ -0,0 +1,183 @@
+#!/bin/bash
+#
+# Copyright 2015 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eu
+
+# Load the test setup defined in the parent directory
+CURRENT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+source "${CURRENT_DIR}/../integration_test_setup.sh" \
+  || { echo "integration_test_setup.sh not found!" >&2; exit 1; }
+
+# Fetch hermetic python and register toolchain.
+function set_up() {
+    cat >>WORKSPACE <<EOF
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+
+http_archive(
+    name = "rules_python",
+    sha256 = "cdf6b84084aad8f10bf20b46b77cb48d83c319ebe6458a18e9d2cebf57807cdd",
+    strip_prefix = "rules_python-0.8.1",
+    url = "https://github.com/bazelbuild/rules_python/archive/refs/tags/0.8.1.tar.gz",
```

Comment:
Fetching this appears to fail on the CI builders, presumably because they're blocking network access for tests.  I'm not really sure how to get around that; without fetching a toolchain from a remote, I guess either those things need to be vendored in to the bazel tree or it's not a hermetic build...

Reply:
I think common practice is to skip the tests for these kind of use cases. An alternative could be mocking all of rules_python, but I don't think that's particularly feasible.

That said, I'm far from a Bazel maintainer, and have the same opinion as you do about not adding tests for these kinds of things.

Reply:
I don't think it's feasible to keep these tests; the automated testing framework cannot run them.
All we can reasonably test is that a provided coverage tool given to py_runtime is used in the way that is expected; a dummy tool could be created and provided for the integration tests.